sdk/keyvault/test-resources.json

```diff
@@ -3,14 +3,80 @@
   "contentVersion": "1.0.0.0",
   "parameters": {
     "baseName": {
-      "type": "String"
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]",
+      "metadata": {
+        "description": "The base resource name."
+      }
     },
     "tenantId": {
-      "type": "String",
-      "defaultValue": "[subscription().tenantId]"
+      "type": "string",
+      "defaultValue": "[subscription().tenantId]",
+      "metadata": {
+        "description": "The tenant ID to which the application and resources belong."
+      }
     },
     "testApplicationOid": {
-      "type": "String"
+      "type": "string",
+      "defaultValue": "b3653439-8136-4cd5-aac3-2a9460871ca6",
+      "metadata": {
+        "description": "The client OID to grant access to test resources."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "The location of the resource. By default, this is the same as the resource group."
+      }
+    },
+    "hsmLocation": {
+      "type": "string",
+      "defaultValue": "centralus",
+      "allowedValues": [
+        "australiacentral",
+        "canadacentral",
+        "centralus",
+        "eastasia",
+        "eastus2",
+        "koreacentral",
+        "northeurope",
+        "southafricanorth",
+        "southcentralus",
+        "southeastasia",
+        "switzerlandnorth",
+        "uksouth",
+        "westeurope",
+        "westus"
+      ],
+      "metadata": {
+        "description": "The location of the Managed HSM. By default, this is 'centralus'."
+      }
+    },
+    "enableHsm": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Whether to enable deployment of Managed HSM. The default is false."
+      }
+    },
+    "enableSoftDelete": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Whether to enable soft delete for the Key Vault. The default is true."
+      }
+    },
```

Comment:
You could actual remove and hard-code this now. It's no longer optional and always true.

Reply:
This is true only for HSMs, correct?

Reply:
No, Key Vault. You can't pass false. It'll err.

Reply:
Though, presumably both. Whatever the case, at least in .NET we don't have any tests for non-soft delete and pretty soon that won't be an option anyway, if not already.